### ISSUE TITLE
[Question] Deploy remote application Jumpserver v3.1.0

### ISSUE THREAD

**Issue by tsukiazuma** [jumpserver/jumpserver#10027]
When I deploy Remote Application Publishers, it usually stops at stages such as Stop Tinker before install, Set Tinkerd on the global system path,... It can't be done in one run or worse, I can't deploy it even though I have to restart my remote device many times.
Please tell me how to do fix it.

**Comment by wojiushixiaobai** [comment]
Use a separate server to deploy the service, and do not deploy the service to a server that is now being used by other services.

Normally, the server must be restarted once.

**Comment by tsukiazuma** [comment]
> Use a separate server to deploy the service, and do not deploy the service to a server that is now being used by other services.
> 
> The server must be restarted once.

Yeah, I used a separate server to deploy. But the result usually looks like this:
![image](https://user-images.githubusercontent.com/75791054/226556036-328cad9d-c25c-4b0a-9dca-3cbb177c26ef.png)

It seems to stop and not execute any more no matter how long it waits.

**Comment by wojiushixiaobai** [comment]
Please complete the Windows Update Service until all patches are installed.

**Comment by tsukiazuma** [comment]
> Please complete the Windows Update Service until all patches are installed.
> 
> The installation log path for Python is:
> 
> ```
> C:\Users\Administrator\AppData\Local\Temp
> ```

it has few other cases like:
![image](https://user-images.githubusercontent.com/75791054/226579864-b913959f-a596-4af0-bc19-225ffd71001c.png)
or stop at the steps: Set Tinkerd on the global system path, Set Tinkerd on the global system path, Check pip packages,...
I tried to wait for hours but it kept stopping there

**Comment by tsukiazuma** [comment]
> Please complete the Windows Update Service until all patches are installed.
> 
> The installation log path for Python is:
> 
> ```
> C:\Users\Administrator\AppData\Local\Temp
> ```

If I check folder, it have stay here:
![image](https://user-images.githubusercontent.com/75791054/226580852-1696d8c5-9c17-4937-b7a4-c54b22767c40.png)

**Comment by wojiushixiaobai** [comment]
- [Stop Tinker before install (jumpserver)](https://github.com/jumpserver/jumpserver/blob/dev/apps/terminal/automations/deploy_applet_host/playbook.yml#L27)
```yaml
    - name: Stop Tinker before install (jumpserver)
      ansible.windows.win_powershell:
        script: |
          if (Get-Service -Name 'JumpServer Tinker' -ErrorAction SilentlyContinue) {
            Stop-Service -Name 'JumpServer Tinker' -Force
          }
          else {
           $Ansible.Changed = $false
          }
```

![image](https://user-images.githubusercontent.com/24326988/226776949-a0fc5261-0ab5-45ff-904e-774852d2578e.png)

Test command from powershell
```powershell
if (Get-Service -Name 'JumpServer Tinker' -ErrorAction SilentlyContinue) {  echo "success" } else { echo "error" }
success
```
![image](https://user-images.githubusercontent.com/24326988/226777478-df229459-6026-480b-a717-de2ec7647b40.png)

**Comment by tsukiazuma** [comment]
I don't think the problem is stop service.
Because when I check, Jumpserver Tinker service don't have.
![image](https://user-images.githubusercontent.com/75791054/226781349-f0afa53d-9d9a-4963-b372-c3a85938ba75.png)
sometimes it can pass this step or sometimes it doesn't.

**Comment by tsukiazuma** [comment]
I will describe in detail.
When I deploy, it can run to install python offline and error.
![image](https://user-images.githubusercontent.com/75791054/226782480-d7cad7bf-bd89-4a2d-884d-755b8915fc54.png)
Then I delete all things such as tinker, python, pip,.. installed.
But when I deploy again, it can look like pause or stop at Gathering Facts,...

**Comment by BaiJiangJie** [comment]
Hi, how are you doing recently? Is the deployment of the remote application publishing machine successful?
@tsukiazuma

**Comment by tsukiazuma** [comment]
> Hi, how are you doing recently? Is the deployment of the remote application publishing machine successful? @tsukiazuma

I have successfully deployed. I'm missing a few patches of Windows Update Service.
If you have a fix, please share it with me
Thanks for your attention.

**Comment by BaiJiangJie** [comment]
Sure, that's great! 

If you encounter any other issues during usage, please feel free to contact us anytime. 

Finally, wish you a happy life！！！
